src/components/CompareResults/beta/ResultsView.tsx

```diff
@@ -31,15 +33,22 @@ function ResultsView(props: ResultsViewProps) {
       backgroundColor: background(themeMode),
     }),
   };
+  const activeComparison: string = useAppSelector(
+    (state: RootState) => state.comparison.activeComparison,
+  );
+  const { switchComparisonData } = useComparison();
   const [searchParams] = useSearchParams();
   const fakeDataParam: string | null = searchParams.get('fakedata');
-
   // TODO: Populate store with real data or fake data pased on URL params
   useEffect(() => {
     if (fakeDataParam === 'true') {
-      dispatch(switchToFakeData());
+      switchComparisonData();
+    }
+    if (fakeDataParam === 'false' || fakeDataParam === null) {
```

Comment:
I'm not sure why you're using a second if block?
```suggestion
    } else {
```